src/helpers/cauldron/lists/getMarketList.ts

```diff
@@ -0,0 +1,140 @@
+import { Contract, providers, utils } from "ethers";
+import { MulticallWrapper } from "ethers-multicall-provider";
+
+import cauldronsConfig from "@/utils/cauldronsConfig";
+import bentoBoxAbi from "@/utils/abi/bentoBox";
+
+import { getMaxToBorrow } from "@/helpers/cauldron/getMaxToBorrow";
+import { getTotalBorrowed } from "@/helpers/cauldron/getTotalBorrowed";
+
+const lensAddress = "0x73f52bd9e59edbdf5cf0dd59126cef00ecc31528";
+import lensAbi from "@/utils/abi/marketLens.js";
+
+type CauldronListItem = {
+  config: object;
+  interest: number;
+  tvl: string;
+  totalBorrowed: string;
+  MIMsLeftToBorrow: string;
+};
+
+export const getMarketList = async (
```

Comment:
Instead of this, can't we just use getCauldronInfo?

Reply:
this helper is designed specifically to prepare for rendering cauldrons for the Cauldrons page - that's why it has that name

I do not know whether it makes sense to request all the information if we do not display it - this style seems to me easier and does not require additional parsing of returned information 

it seemed to me better to unify logic than to make conditions for v2 and v3 and then parse the result (it also works without problems with v1)

Reply:
Not sure... I do know that it's significantly cleaner to do the logic in Solidity and pull all the info at once (versus doing sequential calculations). I also suspect it's overall faster since you only need to make one giant query vs. a bunch in sequence, but we'd have to test it.

Reply:
There isn't that much extra computation here. Mostly fees, which are just constants... Extra seems like it's the borrowFee, liquidationFee, maximumCollateralRatio. Did I miss anything else?

```
export interface MarketInfo {
  borrowFee: BigNumber;
  maximumCollateralRatio: BigNumber;
  liquidationFee: BigNumber;
  interestPerYear: BigNumber;
  marketMaxBorrow: BigNumber;
  userMaxBorrow: BigNumber;
  totalBorrowed: BigNumber;
  oracleExchangeRate: BigNumber;
  collateralPrice: BigNumber;
  totalCollateral: AmountValue;
}
```

Reply:
I think you're right, I updated the helper using lens